lib/code_corps_web/controllers/github_event_controller.ex

```diff
@@ -36,23 +41,31 @@ defmodule CodeCorpsWeb.GithubEventController do
     end
   end
 
-  def create(conn, event_payload) do
-    event_type = conn |> get_event_type()
+  @spec create(Conn.t, map) :: Conn.t
+  def create(%Conn{} = conn, %{} = payload) do
+    type = conn |> get_event_type
+    delivery_id = conn |> get_delivery_id()
+    action = payload |> Map.get("action", "")
 
-    case event_type |> EventSupport.status do
+    case type |> EventSupport.status(action) do
       :supported ->
-        Processor.process_async(event_type, conn |> get_delivery_id, event_payload)
+        Processor.process(fn -> Handler.handle_supported(type, delivery_id, payload) end)
```

Comment:
This is the `CodeCorps.Processor` now, calling `CodeCorps.GitHub.Webhook.Handler` directly